gh-148292: Update ssl._SSLSocket for OpenSSL 4 (#149102)
python/cpython: Misc/NEWS.d/next/Library/2026-04-28-17-47-55.gh-issue-148292.oIq3ml.rst

> :mod:`ssl`: Update :class:`ssl.SSLSocket` for OpenSSL 4. The class now

Reviewer: also mention :class:`ssl.SSLObject` ?

Author: Alright, `ssl.SSObject` is also updated. I completed the NEWS entry.

> OpenSSL 4. The class now remembers if it gets a :exc:`ssl.SSLEOFError`. In this

Author: ```suggestion
OpenSSL 4. The classes now remember if they get a :exc:`ssl.SSLEOFError`. In this
```

> :meth:`~ssl.SSLSocket.read`, :meth:`!sendfile`, :meth:`~ssl.SSLSocket.write`,

Reviewer: :meth:`~ssl.SSLSocket.sendfile` ?

Author: Using :meth:`~ssl.SSLSocket.sendfile` makes the documentation fails to build. It doesn't work since SSLSocket.sendfile() has no anchor at https://docs.python.org/dev/library/ssl.html#ssl.SSLSocket, but SSLSocket.sendfile() method is documented
